content/blog/ggplot2-3-4-0/index.Rmd

```diff
@@ -0,0 +1,220 @@
+
+The switch to `linewidth` goes beyond aesthetics and should target every part of the API that have used `size` to target line width. This is mostly present in theming where `element_rect()` and `element_line()` now uses `linewidth` as argument instead of `size`. As above a deprecation warning will inform you of this change:
+
+```{r}
+ggplot(mtcars) + 
+  geom_point(aes(x = mpg, y = disp)) + 
+  theme(panel.grid = element_line(linewidth = 0.2))
+```
+
+We have done our best to ensure that it is easy for our extension developers to follow the path laid out by ggplot2 when it comes to embracing the new aesthetic, but you will probably experience a period of discrepancy between some of your favorite extensions and ggplot2. I have full confidence that our amazing extension developers will adapt quickly so that period will probably be short.
+
+### On the topic of line width
+
+We have made a few other internal changes when it comes to line widths. The biggest of these are perhaps a new default for polygon line width in `geom_sf()`. The change came about as we already had induced visual changes to old code due to the `linewidth` aesthetic introduction and based on feedback from the spatial community we saw that `size` was most often used to thin the polygon borders. The new default is 0.2 (down from 0.5) and hopefully strikes a nice balance:
+
+```{r}
+nc <- sf::st_read(system.file("shape/nc.shp", package = "sf"), quiet = TRUE)
+p1 <- ggplot(nc) +
+  geom_sf(linewidth = 0.5) + 
+  ggtitle("Old default")
+
+p2 <- ggplot(nc) +
+  geom_sf() + 
+  ggtitle("New default")
+
+p1/p2
+```
+
+More minor is a small fix we did to `guide_colorbar()` where it was brought to our attention that the `ticks.linewidth` and `frame.linewidth` weren't given in the same unit as every other line width in ggplot2. This has been corrected and the default has been adjusted to retain the same look but if you have given these specifically in your code you are likely to notice a visual change.
+
+## Other breaking changes
+
+In the grab-bag of breaking changes we have now formally deprecated `qplot()`. It will continue to work as always but will be a bit noisy about it. Don't expect the function to disappear, but the deprecation signals that we don't intend to do further work on `qplot()` to keep it current with new features etc. In the same vein, `stat()` and `..var..` for marking aesthetics from stats are also formally deprecated in favor of `after_stat()`. Again, the result is that using these old APIs will be noisy but still work.
+
+On the topic of `after_stat()` the values and computations happening inside of this now works on the un-transformed variables rather than the transformed ones. This is a bit esoteric and only applies to aesthetics that have had a scale transformation applied to them, so you may never notice.
```

Comment:
```suggestion
On the topic of `after_stat()`, the values and computations inside of now use the un-transformed variables rather than the transformed ones. This is a bit esoteric and only applies to aesthetics that have had a scale transformation applied to them, so you may never notice.
```